runtime/src/startup.rs

```diff
@@ -52,14 +55,24 @@ macro_rules! stack_size {
 // from start, which is written directly in assembly.
 #[no_mangle]
 extern "C" fn rust_start() -> ! {
-    // TODO: Call memop() to inform the kernel of the stack and heap sizes +
-    // locations. Also, perhaps we should support calling a heap initialization
+    // TODO: Call memop() to inform the kernel of the stack and size +
+    // locations (for debugging).
+    // Also, perhaps we should support calling a heap initialization
     // function?
 
     extern "Rust" {
+        // This function is created by the set_main!() macro.
         fn libtock_unsafe_main() -> !;
     }
+    // TODO: Provide mechanism for dynamic heap size
+    let app_heap_size: usize = 1024;
     unsafe {
+        // TODO: Replace with non-raw syscalls once memop
+        // implemented.
+
+        // Tell the kernel the new app heap break.
+        TockSyscalls::syscall2::<5>([1u32.into(), app_heap_size.into()]);
```

Comment:
This should be operation 11, not operation 1.

Reply:
I mostly based this code on https://github.com/tock/libtock-rs/blob/master/core/src/entry_point/mod.rs#L126 . For libtock1, it looks like we call `set_brk` regardless of whether the alloc feature is in use, so I copied that approach. But shouldn't this be operation 0, not 11? (https://github.com/tock/libtock-rs/blob/022984f9adf1469ba7f53f8d4a7911ea08abe799/core/src/memop.rs#L13 )

Reply:
Hmm, there are two relevant operations here:

1. Setting the process break, which is done via memop 0.
2. Telling the kernel where the heap starts for debugging purposes, which is done via memop 11.

I assumed Alyssa was trying to tell the kernel the heap start (number 2).

In libtock1, setting the process break (number 1) happens twice: once in `_start` where it is set to the end of `.bss` and again in `rust_start` where it is set to the end of the heap. libtock2 is the same, except the second set belongs in the allocator crate rather than in `rust_start`, so it should be in a separate crate.

Reply:
I've confirmed it works without `app_heap_size`.